diff --git a/Encoder/XmlEncoder.php b/Encoder/XmlEncoder.php
index 5ce58fb..41f842a 100644
--- a/Encoder/XmlEncoder.php
+++ b/Encoder/XmlEncoder.php
@@ -23,6 +23,8 @@ use Symfony\Component\Serializer\Exception\UnexpectedValueException;
  */
 class XmlEncoder extends SerializerAwareEncoder implements EncoderInterface, DecoderInterface, NormalizationAwareInterface
 {
+    public const CONTEXT_CDATA_DISABLE = 'xml_cdata_disable';
+
     /**
      * @var \DOMDocument
      */
@@ -462,6 +464,13 @@ class XmlEncoder extends SerializerAwareEncoder implements EncoderInterface, Dec
      */
     private function needsCdataWrapping($val)
     {
+        if (
+            array_key_exists(self::CONTEXT_CDATA_DISABLE, $this->context) &&
+            $this->context[self::CONTEXT_CDATA_DISABLE] === true
+        ) {
+            return false;
+        }
+
         return 0 < preg_match('/[<>&]/', $val);
     }

